xbmc/network/Socket.cpp

```diff
@@ -52,25 +52,34 @@ bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
         m_addr = CAddress("::");
 
         SOCKET testSocket = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
-        setsockopt(testSocket, IPPROTO_IPV6, IPV6_V6ONLY, &zero, sizeof(&zero));
-        // Try to bind a socket to validate ipv6 status
-        for (m_iPort = port; m_iPort <= port + range; ++m_iPort)
+        if (testSocket != INVALID_SOCKET)
         {
-          m_addr.saddr.saddr6.sin6_port = htons(m_iPort);
-          if (bind(testSocket, (struct sockaddr*)&m_addr.saddr, m_addr.size) >= 0)
+          setsockopt(testSocket, IPPROTO_IPV6, IPV6_V6ONLY, &zero, sizeof(&zero));
+          // Try to bind a socket to validate ipv6 status
+          for (m_iPort = port; m_iPort <= port + range; ++m_iPort)
           {
-            m_ipv6Socket = true;
-            break;
+            m_addr.saddr.saddr6.sin6_port = htons(m_iPort);
+            if (bind(testSocket, static_cast<struct sockaddr*>(&m_addr.saddr), m_addr.size) >= 0)
```

Comment:
About them comes this on build:
```
06:14:02 F:\builds\workspace\WIN-32\xbmc\network\Socket.cpp(62): error C2440: 'static_cast': cannot convert from 'SOCKETS::CAddress::<unnamed-type-saddr> *' to 'sockaddr *' [F:\builds\workspace\WIN-32\kodi-build\libkodi.vcxproj]
06:14:02   F:\builds\workspace\WIN-32\xbmc\network\Socket.cpp(62): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
06:14:02 F:\builds\workspace\WIN-32\xbmc\network\Socket.cpp(62): error C2660: 'bind': function does not take 2 arguments [F:\builds\workspace\WIN-32\kodi-build\libkodi.vcxproj]
06:14:02   C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\um\winsock2.h(1647): note: see declaration of 'bind' (compiling source file F:\builds\workspace\WIN-32\xbmc\network\Socket.cpp)
```

Since it unfortunately takes a `union` out of the class, a` reinterpred_cast` will probably be necessary.